### ISSUE TITLE
 java.lang.VerifyError: Bad access to protected data in getfield

### ISSUE THREAD

**Issue by guohao** [square/wire#2063]
I am a green hand on wire and kt, and I creates a simple project to learn wire, but strange exception occurred.

I am trying to solve it but has no idea and it stopped me over 2 days.

The project is here: https://github.com/guohao/wire-study

Anybody can help me ? Thanks!

```
/Users/guohao02/.sdkman/candidates/java/current/bin/java -javaagent:/Applications/IntelliJ IDEA.app/Contents/lib/idea_rt.jar=61986:/Applications/IntelliJ IDEA.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/guohao02/github/wire-study/build/classes/kotlin/main:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.5.30/d68efdea04955974ac1020f8f66ef8176bfbce1f/kotlin-stdlib-1.5.30.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/com.squareup.wire/wire-grpc-client-jvm/4.0.0-alpha.11/210ff4aa0dc0dfe80b9ea47ee24fb0b77244abc2/wire-grpc-client-jvm-4.0.0-alpha.11.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/com.squareup.wire/wire-runtime-jvm/4.0.0-alpha.11/4202472973290ec5c1c16b9f3f52eb354539ef18/wire-runtime-jvm-4.0.0-alpha.11.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.5.30/649ffab7767038323fec0cc41e2d7b0a8f65a378/kotlin-stdlib-common-1.5.30.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/com.squareup.okhttp3/okhttp/4.9.1/51215279c3fe472c59b6b7dd7491e6ac2e28a81b/okhttp-4.9.1.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.5.20/636c5653641cd956de9aee5792155b07ea49825e/kotlin-stdlib-jdk8-1.5.20.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/com.squareup.okio/okio/3.0.0-alpha.9/b1d6cc65ba9a6570fa636011f4c06839b0ca3cab/okio-jvm-3.0.0-alpha.9.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.5.20/218b60e1d446d1e0a18bc7aa8663634b136fbcc5/kotlin-stdlib-jdk7-1.5.20.jar:/Users/guohao02/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlinx/kotlinx-coroutines-core-jvm/1.3.9/4be434f5e86c1998a273e7f19a7286440894f0b0/kotlinx-coroutines-core-jvm-1.3.9.jar io.github.gh.StartKt
Exception in thread "main" java.lang.VerifyError: Bad access to protected data in getfield
Exception Details:
  Location:
    io/github/gh/HelloRequest.hashCode()I @4: getfield
  Reason:
    Type top (current frame, stack[1]) is not assignable to 'io/github/gh/HelloRequest'
  Current Frame:
    bci: @4
    flags: { }
    locals: { 'io/github/gh/HelloRequest' }
    stack: { integer }
  Bytecode:
    0000000: 2ac0 0004 b400 4f3c 1b9a 0045 2ab6 0044
    0000010: b600 513c 1b10 2568 2ab4 001d 4d2c c700
    0000020: 0703 a700 092c b600 543e 1d60 3c1b 1025
    0000030: 682a b400 1f4d 2cc7 0007 03a7 0009 2cb6
    0000040: 0054 3e1d 603c 2ac0 0004 1bb5 004f 1bac
    0000050:                                        
  Stackmap Table:
    full_frame(@37,{Object[#2],Integer,Object[#83]},{Integer})
    full_frame(@43,{Object[#2],Integer,Object[#83]},{Integer,Integer})
    same_locals_1_stack_item_frame(@62,Integer)
    full_frame(@68,{Object[#2],Integer,Object[#83]},{Integer,Integer})
    chop_frame(@78,1)

	at io.github.gh.GrpcGreeterClient.SayHello(GrpcGreeterClient.kt:14)
	at io.github.gh.StartKt$main$1.invokeSuspend(Start.kt:15)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:84)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.github.gh.StartKt.main(Start.kt:9)
	at io.github.gh.StartKt.main(Start.kt)

Process finished with exit code 1

```

**Comment by guohao** [comment]
Degrade wire to 3.7.0 works

**Comment by swankjesse** [comment]
This is a regression in Kotlin 1.5.30 that's fixed in Kotlin 1.5.31.
https://youtrack.jetbrains.com/issue/KT-48295
